app/components/table_component/table_component_controller.js

```diff
@@ -5,23 +5,23 @@ import DataTable from 'datatables.net-dt';
 // Connects to data-controller="table-component"
 export default class extends Controller {
     static values = {
-        sort: Boolean,
-        defaultsortcolumn: String,
+        sortcolumn: String,
         paging: Boolean,
-        searching: Boolean
+        searching: Boolean,
+        noinitsort: Boolean
     }
     connect(){    
         let table_component
         table_component = this.element.childNodes[1]
         let default_sort_column
-        default_sort_column = parseInt(this.defaultsortcolumnValue, 10)
-        if (this.sortValue || this.searchingValue){
+        default_sort_column = parseInt(this.sortcolumnValue, 10)
```

Comment:
I didn't use the number type, cause if the user keeps it empty 'nil' it will be auto intitialized to 0, and the table will be always sorted according to the first column.